extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/handler/FSInternetRadioHandler.java

```diff
@@ -31,29 +32,30 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import org.apache.commons.lang.StringUtils;
-
 /**
  * The {@link FSInternetRadioHandler} is responsible for handling commands, which are
  * sent to one of the channels.
  *
  * @author Patrick Koenemann - Initial contribution
  * @author Mihaela Memova - removed the unused boolean parameter, changed the check for the PIN
+ * @author Svilen Valkanov - changed handler initialization
  */
 public class FSInternetRadioHandler extends BaseThingHandler {
 
     private Logger logger = LoggerFactory.getLogger(FSInternetRadioHandler.class);
 
     private FrontierSiliconRadio radio;
 
+    private boolean isLoginSuccessful = false;
```

Comment:
I'm just thinking of whether it would be better to expose radio->conn->isLoggedIn somehow nicely instead of maintaining our own state here. It'll always be a struggle to keep those two in sync.